### PR TITLE
Intake roller with encoder

### LINKED ISSUE #110: Have barf reset ball count

### PR DESCRIPTION
**Pull request by maxh65000**
Fixes #110

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ private void updateSmartDashboard() {
 
     SmartDashboard.putBoolean("Has Vision", result.HasResult);
     if (result.HasResult) {
-      mRobotLogger.info(Double.toString(result.distance));
+      // mRobotLogger.info(Double.toString(result.distance));
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Robot.java]: Delete commented code, unless you need it for debugging

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -195,8 +197,14 @@ public void teleopInit() {
 
     mStorage.init();
     mDrive.init();
-
-    Config.getInstance().reload();
+    
+    // updated in Intake.java
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Robot.java:200]: Why are these in both Robot.java and Intake.java?

File: src/main/java/frc/robot/subsystems/Storage.java [hunk 3]
```diff
@@ -50,29 +43,30 @@ private Storage() {
 
     mTopRoller.setNeutralMode(NeutralMode.Brake);
     mBottomRoller.setNeutralMode(NeutralMode.Brake);
-
-    mIntakeSensor = new DigitalInput(INTAKE_SENSOR_PORT);
   }
 
-  public void init() {}
+  int getEncoder(){
+    return -mBottomRoller.getSelectedSensorPosition();
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Storage.java:51]: Add a comment to clearly state that the encoder returns negative values for intake motion.

File: src/main/java/frc/robot/subsystems/Storage.java [hunk 4]
```diff
@@ -50,29 +43,30 @@ private Storage() {
...
-      // if the last time we checked the line was broken and now it isn't, we've just stored a ball
-    } else {
-      mWasLineBroke = isLineBroken();
+    int encoderDistance = mStartingEncoderPosition - getEncoder();
+    System.out.println("Got sensor position: " + encoderDistance + " > " + BALL_DISTANCE_IN_ENCODER_TICKS);
+    SmartDashboard.putNumber("Encoder Distance", encoderDistance);
+    
+    // if we've hit our encoder distance target
+    if(encoderDistance >= BALL_DISTANCE_IN_ENCODER_TICKS){
+      // System.out.println("The sensor moved more than the position!");
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Storage.java]: Delete commented code

File: src/main/java/frc/robot/Robot.java [hunk 5]
```diff
@@ -52,7 +52,7 @@
 
   private final int AUTONOMOUS_BALL_COUNT = 3;
   private final double FIRE_DURATION_SECONDS = 0.5;
-  private final int BARF_TIMER_DURATION = 5;
+  private final int BARF_TIMER_DURATION = 3;
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Robot.java:55]: Please reset the storage count on BARF operation.

**Review by jhinspeter — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java (hunk 5 above)
Comment (maxh65000) [reply src/main/java/frc/robot/Robot.java:55]: Fixed

File: src/main/java/frc/robot/subsystems/Storage.java (hunk 4 above)
Comment (maxh65000) [reply src/main/java/frc/robot/subsystems/Storage.java]: Fixed

File: src/main/java/frc/robot/subsystems/Storage.java (hunk 3 above)
Comment (maxh65000) [reply src/main/java/frc/robot/subsystems/Storage.java:51]: Fixed

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (maxh65000) [reply src/main/java/frc/robot/Robot.java:200]: They are initially set in Robot.java but updated in Intake

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (maxh65000) [reply src/main/java/frc/robot/Robot.java]: Fixed

File: src/main/java/frc/robot/subsystems/Storage.java [hunk 6]
```diff
@@ -104,6 +104,10 @@ public void storeBall() {
     mTopRoller.set(ControlMode.PercentOutput, STORE_SPEED);
   }
 
+  public void emptyBalls(){
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Storage.java]: This can be done by calling preloadBalls(0).

**Review by jhinspeter — APPROVED** [review]

**Pull request merged**